### DISCUSSION TITLE
Merge duplicates

### DISCUSSION THREAD

**Discussion by FelixMalfait** [twentyhq/twenty#5244] [Ideas]
We've got a basic feature to surface duplicates but we should build more around it: button to merge duplicates, edit duplicate match criteria...

**Comment by FelixMalfait** [comment]
See also: #6276

**Reply by sidharthramesh** [reply]
Is this currently possible using the API, or by making changes in the backend? If so how? We really need this as a feature.

**Comment by raphspeaks** [comment]
@FelixMalfait I think most people could get around this feature by exposing a “deletedAt” Column on the export. As long as folks have the uniqueID deduping can be done relatively easily in excel/sheets. Once the contacts are merged/deduped then the user could just import the changes and eliminating contacts by setting a date and time on the deletedAt column. As of now we’re unable to do that because the column isn’t exposed.

**Reply by FelixMalfait** [reply]
Hey that's smart, wouldn't have thought of it!
I checked and it seems the deletedAt column is already exposed in the CSV import?

<img width="797" alt="Screenshot 2025-01-01 at 14 00 56" src="https://github.com/user-attachments/assets/976bbea7-d4d1-41ef-837b-36591b6fff15" />

But we just introduced a UI bug that makes it ugly/hard to navigate https://github.com/twentyhq/twenty/issues/9297 - we'll push a fix this week sorry

**Comment by denisrasulev** [comment]
Would it be possible to extend this request and merge not only duplicates, but any number of records I select? For example, in Apollo, one can select multiple records and choose Master record. In another CRM I used you could even select which fields from which record will go to the Master (final record). This will be super convenient. Thanks.

**Reply by FelixMalfait** [reply]
Yes good feedback thanks, when we build this we should enable this

**Comment by themiguelamador** [comment]
This is a must. With email automation, is common to have multiple domains for the same entity, so combining all in the same company, or the same contact, allow to keep things organized. Becomes hard to track when the relationships are scatter arround multiple duplicate items.

**Comment by FelixMalfait** [comment]
Closing as the PR is ongoing, this will be live soon: https://github.com/twentyhq/twenty/pull/13436 

@neo773 will then do a second PR to add a merge button above the duplicate section on record details page. 

Also the criteria for detections are a bit naive so we should consider exposing them (they are already in the DB not in code) and improving them. But it is likely not something we will do in 2025
